test/conftest.py

```diff
@@ -55,3 +55,12 @@ def _connect(demo: gr.Blocks, serialize=True):
             demo.server.thread.join(timeout=1)
 
     return _connect
+
+
+@pytest.fixture(autouse=True)
+def gradio_temp_dir(monkeypatch, tmp_path):
+    """tmp_path is unique to each test function.
+    It will be cleared automatically according to pytest docs: https://docs.pytest.org/en/6.2.x/reference.html#tmp-path
+    """
+    monkeypatch.setenv("GRADIO_TEMP_DIR", str(tmp_path))
+    yield tmp_path
```

Comment:
Nit: do we need `yield` here? The monkeypatch should automatically revert the environmental variable after the test is complete
```suggestion
    return tmp_path
```